src/bp/core/logger/db-persister.ts

```diff
@@ -63,8 +64,19 @@ export class LoggerDbPersister {
       return
     }
 
+    const rows = this.batch.map(log => {
+      const { botId, scope, metadata, message, level, timestamp } = log
+      return {
+        botId,
+        scope,
+        metadata,
+        message,
+        level,
+        timestamp: this.knex.date.format(timestamp)
+      }
```

Comment:
```js 
log => ({ 
 ...log,
 timestamp: this.knex.date.format(log.timestamp)
})
```

There's a possible performance issue here. Let's say you've got 10 000 logs to insert and the batch size is 100, you are gonna loop through the 10 000 logs to map them, then insert 100, then restart the whole process.

Also, unrelated to your changes, there's a double insert issue possible. Check the runTask of `event-collector.ts`, the batch should be spliced before it's inserted (it would also fix the perf issue doing it that way)